### ISSUE TITLE
Spatial jobs are failing

### ISSUE THREAD

**Issue by briandrawert** [StochSS/stochss#397]
```
Stdout:
Got args: ['/home/ubuntu/stochss/pyurdme/pyurdme_wrapper.py', 'output/0b37e45c-ae51-4b31-b988-a2d192cb7d91/0b37e45c-ae51-4b31-b988-a2d192cb7d91.xml', 'output/0b37e45c-ae51-4b31-b988-a2d192cb7d91/results', 'nsm', '1', '1529376287']
Loading model: input_file=output/0b37e45c-ae51-4b31-b988-a2d192cb7d91/0b37e45c-ae51-4b31-b988-a2d192cb7d91.xml
Error unpickling model, could not recreate the mesh.
```
```
Stderr:
Traceback (most recent call last):
  File "/home/ubuntu/stochss/pyurdme/pyurdme_wrapper.py", line 25, in 
    model = pickle.load(open(input_file))
  File "/usr/lib/python2.7/pickle.py", line 1378, in load
    return Unpickler(file).load()
  File "/usr/lib/python2.7/pickle.py", line 858, in load
    dispatch[key](self)
  File "/usr/lib/python2.7/pickle.py", line 1217, in load_build
    setstate(state)
  File "/home/ubuntu/stochss/pyurdme/../app/lib/pyurdme-stochss/pyurdme/pyurdme.py", line 157, in __setstate__
    raise e
TypeError: 'URDMEMesh' object has no attribute '__getitem__'
```

**Comment by ahellander** [comment]
I have seen this error  when the standard pickle module was used. Are you using ‘dill’?

from dill import pickle

/Andreas
-------------------------------------------
Andreas Hellander
Associate Professor, Scientific Computing
Division of Scientific Computing
Department of Information Technology
Uppsala University

On May 23, 2015, at 12:39 AM, Brian Drawert <notifications@github.com<mailto:notifications@github.com>> wrote:


Stdout:
Got args: ['/home/ubuntu/stochss/pyurdme/pyurdme_wrapper.py', 'output/0b37e45c-ae51-4b31-b988-a2d192cb7d91/0b37e45c-ae51-4b31-b988-a2d192cb7d91.xml', 'output/0b37e45c-ae51-4b31-b988-a2d192cb7d91/results', 'nsm', '1', '1529376287']
Loading model: input_file=output/0b37e45c-ae51-4b31-b988-a2d192cb7d91/0b37e45c-ae51-4b31-b988-a2d192cb7d91.xml
Error unpickling model, could not recreate the mesh.


Stderr:
Traceback (most recent call last):
  File "/home/ubuntu/stochss/pyurdme/pyurdme_wrapper.py", line 25, in
    model = pickle.load(open(input_file))
  File "/usr/lib/python2.7/pickle.py", line 1378, in load
    return Unpickler(file).load()
  File "/usr/lib/python2.7/pickle.py", line 858, in load
    dispatch[key](self)
  File "/usr/lib/python2.7/pickle.py", line 1217, in load_build
    setstate(state)
  File "/home/ubuntu/stochss/pyurdme/../app/lib/pyurdme-stochss/pyurdme/pyurdme.py", line 157, in __setstate__
    raise e
TypeError: 'URDMEMesh' object has no attribute '__getitem__'

**Comment by dnath** [comment]
I think this might be because python-libsbml was not installed in older amis ... try this ami : ami-23e90d48

**Comment by briandrawert** [comment]
Confirmed!  On the develop branch, the 'ami-23e90d48'.  Commit: 70f7ab3df4717a9025955815797ebdb718bd4907

**Comment by ahellander** [comment]
Why are spatial jobs affected by libsbml? It is not used anywhere in pyurdme.

/Andreas
-------------------------------------------
Andreas Hellander
Associate Professor, Scientific Computing
Division of Scientific Computing
Department of Information Technology
Uppsala University

On May 25, 2015, at 7:38 PM, Brian Drawert <notifications@github.com<mailto:notifications@github.com>> wrote:


Confirmed! On the develop branch, the 'ami-23e90d48'. Commit: 70f7ab3<https://github.com/StochSS/stochss/commit/70f7ab3df4717a9025955815797ebdb718bd4907>

**Comment by briandrawert** [comment]
It is unclear if the inclusion of libsbml is really the problem or not, but the new AMI correctly simulates spatial jobs which is the bug were were attempting to fix.

**Comment by bbbales2** [comment]
Fixed

**Comment by dnath** [comment]
You should update the rest of the fields ... Otherwise it'll be confusing
which ami we are talking about

On Tuesday, May 26, 2015, bbbales2 <notifications@github.com> wrote:

> Closed #397 <https://github.com/StochSS/stochss/issues/397>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StochSS/stochss/issues/397#event-314053031>.
>


-- 
Sent from Gmail Mobile
